internal/model/rwfolder.go

```diff
@@ -109,6 +109,13 @@ func newRWFolder(m *Model, shortID uint64, cfg config.FolderConfiguration) *rwFo
 	}
 }
 
+// Helper function to check whether either the ignorePerm flag has been 
+// set on the local host or the FlagNoPermBits has been set on the file/dir
+// which is being pulled.
+func (p *rwFolder) ignorePermissions(file protocol.FileInfo) bool {
```

Comment:
I think this is fine since it needs to check `p.ignorePerms` as well which is not accessible from `FileInfo`.